chart/templates/gotk/helm-controller/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/instance: gotk-system
+    app.kubernetes.io/version: latest
+    control-plane: controller
+  name: helm-controller
+  namespace: gotk-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: helm-controller
+  template:
+    metadata:
+      annotations:
+        prometheus.io/port: "8080"
+        prometheus.io/scrape: "true"
+      labels:
+        app: helm-controller
+    spec:
+      containers:
+      - args:
+        - --events-addr=
+        - --watch-all-namespaces=true
+        - --log-level=info
+        - --log-json
+        - --enable-leader-election
+        env:
+        - name: RUNTIME_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        {{- if .Values.gotk.helm-controller.proxy }}
+        - name: HTTPS_PROXY
+          value: {{ .Values.global.env.httpsProxy }}
+        - name: NO_PROXY
+          value: {{ .Values.global.env.noProxy }}
+        {{- end }}
+        {{- if .Values.gotk.helm-controller.image.registry }}
+        image: {{ .Values.gotk.helm-controller.image.registry }}/{{ .Values.gotk.helm-controller.image.repository | default "fluxcd" }}/helm-controller:{{ .Values.gotk.helm-controller.image.tag }}
+        {{- else }}
+        image: "ghcr.io/{{ .Values.gotk.helm-controller.image.repository | default "fluxcd" }}/helm-controller:{{ .Values.gotk.helm-controller.image.tag }}"
+        {{- end }}       
```

Comment:
The convention for setting up image info is typically the following:

```yaml
  image: {{ .Values.gotk.helm-controller.image.repository }}:{{ .Values.gotk.helm-controller.image.tag }}
```

The above assumes that the image will always be "helm-controller". Also, no need to have a registry property. It's always part of the image name. Removing the registry prop allows your templating logic to not need an "if" statement.